### ISSUE TITLE
Why training only 1 class at a time ? 

### ISSUE THREAD

**Issue by joihn** [marcoaversa/diffinfinite#9]
Hello, thanks for this repo, 
about this line: 
```
mask=torch.where(mask==self.subclasses[self.tmp_index-1], self.tmp_index, 0)
```
https://github.com/marcoaversa/diffinfinite/blob/master/dataset.py#L353

**If I understand correctly:**
The guidance mechanism used here trains only 1 class at a time
In other word, a mask can contains only zeros (unkown class) and 1 other class (specifically selected for this sample)
If there are other class in this mask, they are sent to 0

**What is the reason behind this ?** 
Why can't a mask contains multiple class during training ? 
for example class `idx_1` on the righthalf, class `idx_2` on the left half ? 

Making proper use of all the label pixel available at each iteration would significantly speed up training no ?

**Comment by joihn** [comment]
Hum, maybe it's for minimizing intra-mask class imbalance influence ?

**Comment by bhosalems** [comment]
that is how a method is set I think, if we see the generation pipeline, it predicts noise in a training iteration for every class before adding them all together.
